db/fixtures/ae_datastore/ManageIQ/System/CommonMethods/QuotaMethods.class/__methods__/limits.rb

```diff
@@ -0,0 +1,58 @@
+#
+# Description: Get quota values.
+#
+
+def quota_values(model_attribute, tag_name)
+  $evm.log(:info, "Getting Quota Values for Model: #{model_attribute} Tag Name: #{tag_name}")
+  object_value = quota_model_value(model_attribute)
+  tag = @source.tags(tag_name).first
+  tag_value = tag_value(tag_name, tag) || 0
+  final_quota_value(tag_value, object_value)
+end
+
+def quota_model_value(model_attr)
+  parent_model_value(model_attr)
+end
+
+def final_quota_value(tag_quota_value, object_quota_value)
+  val = tag_quota_value.zero? ? object_quota_value : tag_quota_value
+  $evm.log(:info, "Final Quota Value: #{val}")
+  val
+end
+
+def vmdb_object(object, id)
+  $evm.vmdb(object).find_by_id(id)
+end
+
+def parent_model_value(attr)
+  value = $evm.parent[attr].to_i
+  $evm.log(:info, "Quota Model #{attr}: #{value}") unless value.zero?
+  value
+end
+
+def tag_value(tag, tag_value)
+  value = tag_value.to_i
+  $evm.log(:info, "Quota Tag #{tag}: #{value}") unless value.zero?
+  value
+end
+
+def get_option_value(request, option)
+  request.get_option(option).to_i
+end
+
+def determine_quota_value(model_attribute, tag_name)
+  $evm.log(:info, "Determine Quota Value: #{model_attribute} Tag name: #{tag_name}")
+  quota_values(model_attribute, tag_name)
+end
+
+@source = $evm.root['quota_source']
+quota_max = {}
+quota_warn = {}
+
+%w(storage vms cpu memory).each do |i|
```

Comment:
@tinaafitz 
We should move these into a constant and put it at the top of the file, would make it easy to add new types of quotas to check.